inference-engine/tests/functional/plugin/shared/src/single_layer_tests/tensor_iterator.cpp

```diff
@@ -0,0 +1,230 @@
+                tensor_iterator->get_iter_value(results[0]);
+
+                // 3. Outer function
+                function = std::make_shared<ngraph::Function>(ngraph::OutputVector{tensor_iterator->output(0), tensor_iterator->output(1)}, outer_params);
+                break;
+            }
+            case TensorIteratorBody::RNN: {
+                inputShapes = {{batch, seq_lenghts, input_size},
+                               {batch,       hidden_size},
+                               {hidden_size, input_size},
+                               {hidden_size, hidden_size},
+                               {hidden_size}};
+                auto outer_params = ngraph::builder::makeParams(ngPrc, {inputShapes[0], inputShapes[1]});
+
+                // 1. Create TensorIterator body.
+                inputShapes[0][1] = 1; // sliced dimension
+                auto body_params = ngraph::builder::makeParams(ngPrc, {inputShapes[0], inputShapes[1]});
+                std::vector<ngraph::Shape> WRB = {inputShapes[2], inputShapes[3], inputShapes[4]};
+                auto squeeze = std::make_shared<ngraph::opset5::Squeeze>(body_params[0], axis);
+                ngraph::OutputVector out_vector = {squeeze, body_params[1]};
+                auto rnn_cell = ngraph::builder::makeRNN(out_vector, WRB, hidden_size, {"tanh"}, {}, {}, clip);
+                auto unsqueeze = std::make_shared<ngraph::opset5::Unsqueeze>(rnn_cell->output(0), axis);
+                ngraph::ResultVector results{std::make_shared<ngraph::opset1::Result>(rnn_cell),
+                                             std::make_shared<ngraph::opset1::Result>(unsqueeze)};
+                auto body = std::make_shared<ngraph::Function>(results, body_params, "rnn_cell");
+                tensor_iterator->set_function(body);
+
+                // 2. Set PortMap
+                if (direction == ngraph::op::RecurrentSequenceDirection::FORWARD) {
+                    tensor_iterator->set_sliced_input(body_params[0], outer_params[0], 0, 1, 1, -1, 1);
+                    tensor_iterator->get_concatenated_slices(results[1], 0, 1, 1, -1, 1);
+                } else if (direction == ngraph::op::RecurrentSequenceDirection::REVERSE) {
+                    tensor_iterator->set_sliced_input(body_params[0], outer_params[0], -1, -1, 1, 0, 1);
+                    tensor_iterator->get_concatenated_slices(results[1], -1, -1, 1, 0, 1);
+                } else {
+                    NGRAPH_CHECK(false, "Bidirectional case is not supported.");
+                }
+
+                tensor_iterator->set_invariant_input(body_params[1], outer_params[1]);
+                tensor_iterator->get_iter_value(results[0]);
+
+                // 3. Outer function
+                function = std::make_shared<ngraph::Function>(ngraph::OutputVector{tensor_iterator->output(0), tensor_iterator->output(1)}, outer_params);
+                break;
+            }
+        }
+        if (should_decompose) {
```

Comment:
We should do it in `SetUp()`? I offer to move it in test_body

Reply:
It should be done after the creation of ngraph::function, I guess it a suitable place

Reply:
`SetUp ` is called before the `ngraph::function` creation, I guess it's a suitable place for this code for now